serial/rfc2217.py

```diff
@@ -613,9 +613,7 @@ def read(self, size=1):
             while len(data) < size:
                 if self._thread is None:
                     raise SerialException('connection failed (reader thread died)')
-                data += self._read_buffer.get(True, timeout.time_left())
-                if timeout.expired():
-                    break
+                data += self._read_buffer.get(True, self._timeout)
```

Comment:
unfortunately, this is not equivalent. The original code uses a special timeout object (that uses a monotonic clock if available).